### PR TITLE
8333099: Missing check for is_LoadVector in StoreNode::Identity

### PR DESCRIPTION
**Pull request by dafedafe**
[JDK-8325520](https://bugs.openjdk.org/browse/JDK-8325520) introduced a check for type equality in `StoreNode::Identity` in the specific case of a load vector followed by a store vector.
Unfortunately the memory node operand might actually not be of type `LoadVector`. So, before retrieving its type, a check for `is_LoadVector` is necessary.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333099](https://bugs.openjdk.org/browse/JDK-8333099): Missing check for is_LoadVector in StoreNode::Identity (**Bug** - P3)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19449/head:pull/19449` \
`$ git checkout pull/19449`

Update a local copy of the PR: \
`$ git checkout pull/19449` \
`$ git pull https://git.openjdk.org/jdk.git pull/19449/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19449`

View PR using the GUI difftool: \
`$ git pr show -t 19449`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19449.diff">https://git.openjdk.org/jdk/pull/19449.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19449#issuecomment-2136917119)

### CODE REVIEW

**Review by chhagedorn — APPROVED** [review]
Looks good! Can you also add a regression test?

**Comment by dafedafe** [comment]
> Looks good! Can you also add a regression test?

Yep, added. Thanks @chhagedorn!

File: test/hotspot/jtreg/compiler/vectorapi/TestIsLoadVector.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package compiler.vectorapi;
+
+/*
+ * @test
+ * @bug 8333099
+ * @summary We should check for is_LoadVector before checking for equality between vector types
+ * @run main/othervm -XX:CompileCommand=compileonly,compiler.vectorapi.TestIsLoadVector::test -Xcomp compiler.vectorapi.TestIsLoadVector
+ */
+
+ public class TestIsLoadVector {
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/vectorapi/TestIsLoadVector.java:33]:
```suggestion
public class TestIsLoadVector {
```

File: test/hotspot/jtreg/compiler/vectorapi/TestIsLoadVector.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      int i18 = 1;
+      while (++i18 < 41) {
+          iArrFld[i18] >>= s;
+      }
+  }
+
+  public static void main(String[] strArr) {
+    for (int i = 0; i < 100; i++) {
+        test();
+    }
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/vectorapi/TestIsLoadVector.java:61]:
```suggestion
      for (int i = 0; i < 100; i++) {
          test();
      }
```

**Review by chhagedorn — COMMENTED** [review]
Thanks for adding a test!

**Review by chhagedorn — APPROVED** [review]

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by dafedafe** [comment]
Thanks for reviewing @chhagedorn @TobiHartmann.

**Pull request closed without merging**